### PR TITLE
[improve][broker] Cancel the loadShedding task when closing pulsar service

### PR DESCRIPTION
**Pull request by gaoran10**
### Motivation

The PR https://github.com/apache/pulsar/pull/16408 supported loadBalancerSheddingIntervalMinutes dynamic configuration, but the scheduled load-shedding task may block pulsar service close operation, the pulsar service need to wait until the scheduled load-shedding task executed or reach broker shutdown timeout. If the load-shedding task interval time and broker shutdown timeout are long, the pulsar service close operation will waste a lot of time.

### Modifications

Cancel the scheduled task when closing the pulsar service.

### Verifying this change

Add a test to verify the pulsar service could be closed gracefully in time.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/LoadSheddingTask.java [hunk 1]
```diff
@@ -52,25 +55,24 @@ public void run() {
         } catch (Exception e) {
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/LoadSheddingTask.java:55]: In the first line of method `run` method, we can add a check to make quick failure:

```java

public void run(){
    if (isCancel){
        return;
    }
    ...
}

```

**Review by poorbarcode — APPROVED** [review]
Good work. 

I left a suggestion

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/LoadSheddingTask.java (hunk 1 above)
Comment (gaoran10) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/LoadSheddingTask.java:55]: Thanks, good idea.

**Review by congbobo184 — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        pulsar.start();
+        log.info("Pulsar started. brokerServiceUrl: {} webServiceAddress: {}", pulsar.getBrokerServiceUrl(),
+                pulsar.getWebServiceAddress());
+        return pulsar;
+    }
+
+    @Test(timeOut = 30_000)
+    public void closeInTimeTest() throws Exception {
+        // The pulsar service is not used, so it should be closed gracefully in short time.
+        pulsar.close();
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java:59]: Can we add a check for the `LoadSheddingTask.java.future` is cancelled, or we can just add a method like `LoadSheddingTask.isCancelled`

Comment (Jason918) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java:59]: +1

Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java:59]: I adjust the test, PTAL

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.concurrent.ScheduledFuture;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.broker.auth.MockedPulsarServiceBaseTest;
+import org.apache.pulsar.broker.loadbalance.LoadSheddingTask;
+import org.awaitility.reflect.WhiteboxImpl;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+@Slf4j
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java:34]: It's better to have a test group name

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+public class PulsarServiceCloseTest extends MockedPulsarServiceBaseTest {
+
+    @BeforeMethod
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java:37]: BeforeClass

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    @Override
+    protected void setup() throws Exception {
+        super.internalSetup();
+    }
+
+    @AfterMethod(alwaysRun = true)
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java:43]: AfterClass

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java (hunk 3 above)
Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java:34]: fix

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java (hunk 4 above)
Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java]: fix

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java (hunk 5 above)
Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/PulsarServiceCloseTest.java]: fix

**Review by Technoboy- — APPROVED** [review]

**Review by coderzc — APPROVED** [review]

**Pull request merged**